cclunit-framework-source/src/main/resources/cclut_xml_access_subs.inc

```diff
@@ -267,20 +267,19 @@ end ;;;cclut::writeXmlElement
 */
 subroutine cclut::getXmlListItemHandle(cclutPElementHandle, cclutPChildName, cclutPOccurrenceIndex)
   if (cclutPElementHandle != 0)
-    declare zero             = h with protect, constant(0)
-    declare cclutChIdx       = h with protect, noconstant(0)
-    declare cclutChCnt       = h with protect, noconstant(0)
-    declare cclutOsNumber    = h with protect, noconstant(0)
+    declare cclutHZero       = h with protect, constant(0)
```

Comment:
This isn't relevant to this specific story; but I figured I'd bring it up.  When I was troubleshooting this h/i4 issue, this was causing me hassle because cclutHZero (zero) is declared inside the conditional, but the return value if cclutPElementHandle is 0 leverages cclutHZero even though it's outside the conditional.  Is this expected to work correctly?  Or is it just because when everything is functioning normally, cclutPElementHandle isn't supposed to be 0, so the happy path causes no problems?

Reply:
That's a mistake. I'll correct it. 

Do you think think the framework should return "F" if there are no qualifying tests? 
I posed this question on the internal support board but got no responses.

Reply:
This would be an "F" to cause "mvn test" to fail, right?

It's a good question.  In other testing frameworks that I've seen, this is usually a warning condition and not an outright error.  Do you feel that outputting to the console (and any logs that are written to /target) would be "enough" to still let the Maven build pass?  

If not (or that it would mostly be glossed over in light of a successful build), then I think it would be reasonable to fail it.  If one is explicitly running the test goal, and there are no tests, that seems like something that needs to be verified by the person in question.

However, to play devil's advocate, if a continuous integration system is set up that's running the same commands against a large number of projects, this would be less than desirable if they had to go set up a "fake test" just for it to run.

My first thought for reconciling those two would be a configuration option, but that may be too much effort to incorporate at the moment (I haven't given any consideration to the scope of such work; nor do I want to presume schedules/timetables).  It would be a non-passive change, so we'd definitely want to update the documentation stating as such, and I think outputting it to the console/log (regardless of failing or not) would be good (I only noticed when I looked in the test output files for a test I was expecting to have failed).

Reply:
> In particular, CCL Unit Tests are "failing" in Provide, because no tests are being found, so Maven builds successfully every time even when there should be failures. 

(from your ucern post) 
The actual problem is that tests were not failing even though some of them should have been failing. 

The framework could certainly return "WARNING" rather than "FAILED", but it would go unnoticed by ccl-maven-plugin. On the other hand cerreal-maven-plugin would choke on "WARNING" and fail the build because that is not one of the enum values listed in the xml schema for test results. Both could be improved to understand "WARNING", of course.

If there is a big reactor build with some projects that have tests and some that do not, one could create two reactor projects, one that only builds the projects having no tests using **mvn compile** and one that builds only the projects having tests using **mvn test**.


I misspoke earlier. I had posed the question in this project not on ucern. I have since posted it on ucern.

Reply:
> (from your ucern post)
The actual problem is that tests were not failing even though some of them should have been failing.

Right, for my specific instance, yes, this would have been beneficial.  I'm just trying to think from more of an abstract perspective if this is the "right" answer.  I think either answer is justifiable and neither approach is definitively "wrong".  I'll be curious to see what others' opinions are from uCern and why they feel that way.